cajita/src/Cajita_SparseIndexSpace.hpp

```diff
@@ -500,23 +503,39 @@ class SparseIndexSpace
 
   private:
     //! block index space, map tile ijk to tile
-    BlockIndexSpace<MemorySpace, cell_bits_per_tile_dim, cell_num_per_tile_dim,
-                    cell_num_per_tile, hash_type, key_type, value_type>
+    BlockMap<MemorySpace, cell_bits_per_tile_dim, cell_num_per_tile_dim,
+             cell_num_per_tile, hash_type, key_type, value_type>
         _block_id_space;
     //! tile index space, map cell ijk to cell local No inside a tile
-    TileIndexSpace<cell_bits_per_tile_dim, cell_num_per_tile_dim,
-                   cell_num_per_tile>
+    TileMap<cell_bits_per_tile_dim, cell_num_per_tile_dim, cell_num_per_tile>
         _tile_id_space;
     //! space size (global), channel size
     Kokkos::Array<int, rank> _min;
     Kokkos::Array<int, rank> _max;
 };
 
+//---------------------------------------------------------------------------//
+// Creation function for SparseMap from GlobalMesh<SparseMesh>
+template <typename MemorySpace, class Scalar,
+          unsigned long long CellPerTileDim = 4,
+          HashTypes Hash = HashTypes::Naive, typename Key = uint64_t,
+          typename Value = uint64_t>
+SparseMap<MemorySpace, CellPerTileDim, Hash, Key, Value> createSparseMap(
```

Comment:
Could return `auto`

Reply:
Yes, that should be a more clean way. 
But I notice that almost all creation functions in Cajita use this way rather than auto, even though they may require lines of code for the return type. 
So shall I follow this coding style here? Or shall I change it to auto?

Reply:
I think this is fine for now